### ISSUE TITLE
AuthenticationError: No code found in redirect

### ISSUE THREAD

**Issue by GePlusE** [iterative/PyDrive2#70]
I have a related issue to [issue #58](https://github.com/iterative/PyDrive2/issues/58) I followed the instructions to set up PyDrive2 within an existing project. After installing PyDrive2 and creating the client_secrets.json (which is in the same directory) I tried to test it ran this code
````
from pydrive2.auth import GoogleAuth
from pydrive2.drive import GoogleDrive

gauth = GoogleAuth()
gauth.LocalWebserverAuth()

drive = GoogleDrive(gauth)
````
after runnig I get this Traceback:
````
Failed to find "code" in the query parameters of the redirect.
Try command-line authentication
Traceback (most recent call last):
  File "/Users/gepluse/CodeProjects/ComunioBuster/test2.py", line 5, in <module>
    gauth.LocalWebserverAuth()
  File "/Users/gepluse/CodeProjects/ComunioBuster/venv/lib/python3.8/site-packages/pydrive2/auth.py", line 125, in _decorated
    code = decoratee(self, *args, **kwargs)
  File "/Users/gepluse/CodeProjects/ComunioBuster/venv/lib/python3.8/site-packages/pydrive2/auth.py", line 273, in LocalWebserverAuth
    raise AuthenticationError("No code found in redirect")
pydrive2.auth.AuthenticationError: No code found in redirect
````

I tried @shcheklein solution by testing it within a clean environment but could get it to work either. The problem seemed to be the application type "web". After creating new credentials with an application type "desktop" the authentication process worked with the new environment.
Unfortunately, although I use the same credentials in my existing project I still get the same Traceback.

Do you have any ideas why this happens?

EDIT:
Now it didn't work in the clean environment either. Is it possible that the authentication process only works one time?

_Originally posted by @GePlusE in https://github.com/iterative/PyDrive2/issues/58#issuecomment-735270057_

**Comment by shcheklein** [comment]
@GePlusE the problem here is that we don't know how to reproduce this - it works for me :(

have you seen this response - https://stackoverflow.com/questions/63955197/pydrive-2-0-authenticationerror-no-code-found-in-redirect ?

how do you run your code? could you create a completely empty virtualenv and run this code from the Python repl?

**Comment by GePlusE** [comment]
@shcheklein thanks for the quick reply. I've seen this, but I am not sure what LucSpan did differently to make it work.
Am I right that all he did was putting it inside a function?

I run my code via VS Code and I already tried a clean environment but with VS Code.

**Comment by GePlusE** [comment]
I don't know if this is helpful but I tried it again (without any changes since my last unsuccessful try yesterday) and now the authentication flow was successful and a file was uploaded to my Google Drive.

Then I tried it with another file but the same error appeared again. The only change I made was the name of the file I tried to upload (yes, the file exists and is in the same directory ;) )

Could it be possible that the authentication can only be done once in a specific period?

I made changes according to this [StackOverflow Solution](https://stackoverflow.com/a/24542604/12554227) by Dano. Maybe the save and load of `mycreds.txt` will help to solve this issue - I will post the result after a next try tomorrow because at the moment I still get the described error.
